src/inmanta/server/services/compilerservice.py

```diff
@@ -444,16 +451,37 @@ async def _run(self, compile: data.Compile) -> None:
             )
         await asyncio.sleep(wait)
 
+        merge_candidates: List[data.Compile] = [
+            c
+            for c in await data.Compile.get_next_compiles_for_environment(compile.environment)
+            if not c.id == compile.id and CompilerService._compile_merge_key(c) == CompilerService._compile_merge_key(compile)
```

Comment:
I would lift the CompilerService._compile_merge_key(compile) out of the loop, as the to_dto() is (I think) an expensive operation. 

This whole code block could probably also become a  single database query, but let's keep that for the future